packages/core/contracts/cross-chain-oracle/chain-adapters/Optimism_ChildMessenger.sol

```diff
@@ -0,0 +1,100 @@
+     * @dev The caller of this function must be the parent messenger, over the canonical bridge.
+     * @param newOracleSpoke address of the new oracle spoke, deployed on L2.
+     */
+    function setOracleSpoke(address newOracleSpoke) public onlyFromCrossDomainAccount(parentMessenger) {
+        oracleSpoke = newOracleSpoke;
+        emit SetOracleSpoke(newOracleSpoke);
+    }
+
+    /**
+     * @notice Changes the stored address of the parent messenger, deployed on L1.
+     * @dev The caller of this function must be the parent messenger, over the canonical bridge.
+     * @param newParentMessenger address of the new parent messenger, deployed on L1.
+     */
+    function setParentMessenger(address newParentMessenger) public onlyFromCrossDomainAccount(parentMessenger) {
+        parentMessenger = newParentMessenger;
+        emit SetParentMessenger(newParentMessenger);
+    }
+
+    /**
+     * @notice Changes the default gas limit that is sent along with transactions to Ethereum.
+     * @dev The caller of this function must be the parent messenger, over the canonical bridge.
+     * @param newDefaultGasLimit the new L1 gas limit to be set.
+     */
+    function setDefaultGasLimit(uint32 newDefaultGasLimit) public onlyFromCrossDomainAccount(parentMessenger) {
+        defaultGasLimit = newDefaultGasLimit;
+        emit SetDefaultGasLimit(newDefaultGasLimit);
+    }
+
+    /**
+     * @notice Sends a message to the parent messenger via the canonical message bridge.
+     * @dev The caller must be the OracleSpoke on L2. No other contract is permissioned to call this function.
+     * @dev The L1 target, the parent messenger, must implement processMessageFromChild to consume the message.
+     * @param data data message sent to the L1 messenger. Should be an encoded function call or packed data.
+     */
+    function sendMessageToParent(bytes memory data) public override {
+        require(msg.sender == oracleSpoke, "Only callable by oracleSpoke");
+        bytes memory dataSentToParent = abi.encodeWithSignature("processMessageFromChild(bytes)", data);
+        sendCrossDomainMessage(parentMessenger, defaultGasLimit, dataSentToParent);
+        emit MessageSentToParent(dataSentToParent, parentMessenger, defaultGasLimit);
+    }
+
+    /**
+     * @notice Process a received message from the parent messenger via the canonical message bridge.
+     * @dev The caller must be the the parent messenger, sent over the canonical message bridge.
+     * @param data data message sent from the L1 messenger. Should be an encoded function call or packed data.
+     * @param target desired recipient of `data`. Target must implement the `processMessageFromParent` function. Having
+     * this as a param enables the L1 Messenger to send messages to arbitrary addresses on the L1. This is primarily
+     * used to send messages to the OracleSpoke and GovernorSpoke on L2.
+     */
+    function processMessageFromParent(bytes memory data, address target)
```

Comment:
Similar comment here as in @nicholaspai's previous PR -- can we call this method something different than the method it calls?

I think it's confusing to have both the ChildMessengerInterface and the ChildMessengerConsumerInterface have the same method name.

Reply:
yup, can do. I changed the functions that call cross-chain to `processMessageFromCrossChainChild` and `processMessageFromCrossChainParent` and left the ones that operate on the "same chain" the same.